SharedSource/UnitTests/Reflection/TypeExtensionsTest.cs

```diff
@@ -0,0 +1,152 @@
+//
+
+using System;
+using System.Reflection;
+using NUnit.Framework;
+using Remotion.Reflection;
+
+#nullable enable
+// ReSharper disable once CheckNamespace
+namespace Remotion.UnitTests.Reflection
+{
+  [TestFixture]
+  public class TypeExtensionsTest
+  {
+    [Test]
+    public void GetAssemblyQualifiedNameSafe_WithSystemType_ReturnsAssemblyQualifiedName ()
+    {
+      var typeArgument = typeof (string);
+
+      Assert.That (typeArgument.AssemblyQualifiedName, Is.Not.Null);
+      Assert.That (typeArgument.GetAssemblyQualifiedNameSafe(), Does.StartWith ("System.String,"));
+    }
+
+    [Test]
+    public void GetAssemblyQualifiedNameSafe_WithGenericTypeParameter_ReturnsTypeName ()
+    {
+      var genericTypeParameters = typeof (CustomType<>).GetTypeInfo().GenericTypeParameters;
+
+      Assert.That (genericTypeParameters[0].AssemblyQualifiedName, Is.Null);
+      Assert.That (genericTypeParameters[0].FullName, Is.Null);
+      Assert.That (genericTypeParameters[0].GetAssemblyQualifiedNameSafe(), Is.EqualTo ("T"));
+    }
+
+    [Test]
+    public void GetAssemblyQualifiedNameChecked_WithSystemType_ReturnsAssemblyQualifiedName ()
+    {
+      var typeArgument = typeof (CustomType<string>).GetGenericArguments();
+
+      Assert.That (typeArgument[0].AssemblyQualifiedName, Is.Not.Null);
+      Assert.That (typeArgument[0].GetAssemblyQualifiedNameChecked(), Does.StartWith ("System.String,"));
+    }
+
+    [Test]
+    public void GetAssemblyQualifiedNameChecked_WithGenericTypeParameter_Throws ()
+    {
+      var genericTypeArgumentsParameters = typeof (CustomType<>).GetTypeInfo().GenericTypeParameters;
+
+      Assert.That (genericTypeArgumentsParameters[0].AssemblyQualifiedName, Is.Null);
+      Assert.That (
+          () => genericTypeArgumentsParameters[0].GetAssemblyQualifiedNameChecked(),
+          Throws.InvalidOperationException.With.Message.EqualTo ("Type 'T' does not have an assembly qualified name."));
+    }
+
+    [Test]
+    public void GetFullNameSafe_WithSystemType_ReturnsFullName ()
+    {
+      var typeArgument = typeof (string);
+
+      Assert.That (typeArgument.FullName, Is.Not.Null);
+      Assert.That (typeArgument.GetFullNameSafe(), Is.EqualTo ("System.String"));
+    }
+
+    [Test]
+    public void GetFullNameSafe_WithGenericTypeParameter_ReturnsTypeName ()
+    {
+      var genericTypeParameters = typeof (CustomType<>).GetTypeInfo().GenericTypeParameters;
+
+      Assert.That (genericTypeParameters[0].FullName, Is.Null);
+      Assert.That (genericTypeParameters[0].GetFullNameSafe(), Is.EqualTo ("T"));
+    }
+
+    [Test]
+    public void GetFullNameChecked_WithSystemType_ReturnsFullName ()
+    {
+      var typeArgument = typeof (string);
+
+      Assert.That (typeArgument.FullName, Is.Not.Null);
+      Assert.That (typeArgument.GetFullNameChecked(), Is.EqualTo ("System.String"));
+    }
+
+    [Test]
+    public void GetFullNameChecked_WithGenericTypeParameter_Throws ()
```

Comment:
```suggestion
    public void GetFullNameChecked_WithGenericTypeParameter_ThrowsInvalidOperationException ()
```

It's not mandatory for simple cases, but good practice to give the type of the exception